Uplift/Views/Onboarding/OnboardingView.swift

```diff
@@ -0,0 +1,177 @@
+//
+//  OnboardingView.swift
+//  Uplift
+//
+//  Created by Phillip OReggio on 11/18/19.
+//  Copyright © 2019 Cornell AppDev. All rights reserved.
+//
+
+import UIKit
+
+class OnboardingView: UIView {
+
+    // MARK: - Private vars
+    private let imageView = UIImageView()
+    private let titleLabel = UILabel()
+    private var tableView: UITableView?
+
+    private var tableData: [String]?
+    private var showingClasses = false
+
+    private let tableViewCellHeight: CGFloat = 60
+    private let tableViewCellSpacing: CGFloat = 14
+
+    // MARK: = Delegation
+    var delegate: (([String]) -> Void)?
+
+    // MARK: - Public vars
+    var favorites: [String] = [] // User's selected favorite gyms/classes
+    var hasTableView = false
+
+    init(image: UIImage, text: String, gymNames: [String]? = nil, classNames: [String]? = nil) {
+        super.init(frame: .zero)
+
+        imageView.image = image
+        imageView.contentMode = .scaleAspectFit
+        addSubview(imageView)
+
+        titleLabel.attributedText = generateAttributedString(for: text)
+        titleLabel.numberOfLines = 3
+        addSubview(titleLabel)
+
+        if let data = gymNames ?? classNames {
+            hasTableView = true
+            tableData = data
+            showingClasses = tableData == classNames
+
+            tableView = UITableView(frame: .zero, style: .plain)
+            tableView?.delegate = self
+            tableView?.dataSource = self
+            tableView?.register(FavoriteCell.self, forCellReuseIdentifier: FavoriteCell.identifier)
+            tableView?.isScrollEnabled = false
+            tableView?.clipsToBounds = false
+            tableView?.separatorStyle = .none
+            self.addSubview(tableView!)
+        }
+
+        setUpConstraints()
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    func getSize() -> CGSize {
+        return hasTableView ?
+            CGSize(width: 388, height: 700) :
+            CGSize(width: 388, height: 269)
+    }
+
+    private func setUpConstraints() {
+        let imageSize = hasTableView ? CGSize(width: 214, height: 183) : CGSize(width: 295, height: 222.3)
+        let labelWidth: CGFloat = 301
+
+        imageView.snp.makeConstraints { make in
+            make.size.equalTo(imageSize)
+            make.top.centerX.equalToSuperview()
+        }
+
+        if let table = tableView { // With Table View
+            let labelHeight: CGFloat = 80//88
+            let labelTopOffset: CGFloat = 174
+
+            let tableViewBottomOffset: CGFloat = 17
+            let tableViewSize = CGSize(width: 388, height: 308)
+
+            titleLabel.snp.makeConstraints { make in
+                make.centerX.equalToSuperview()
+                make.width.equalTo(labelWidth)
+                make.height.equalTo(labelHeight)
+                make.top.equalToSuperview().inset(labelTopOffset)
```

Comment:
should top be relative to the imageview?

Reply:
I did it relative to the top so the label's vertical positioning is consistent when swiping the slides